CWE/events/Essential Events 8.txt

```diff
@@ -10680,27 +10680,29 @@ country_event = {
 	desc = "Years of communist planned economy has wrought much economic destruction to our industries. With the end of the planned economy, our nation's markets have been able to grow through market forces and is on the track to a healthy recovery."
 	picture = "soviet_stagnation_end"
 
-	trigger = {
-	
-		#Stagnation
+    	trigger = {
+    
+     	   #Stagnation
 		OR = { 
 			has_country_modifier = soviet_stagnation_1 
 			has_country_modifier = soviet_stagnation_2 
-			has_country_modifier = soviet_stagnation_3 
-			has_country_modifier = soviet_stagnation_4 
+            		has_country_modifier = soviet_stagnation_3 
+           	 	has_country_modifier = soviet_stagnation_4 
 		}
-		
+        
 		OR = {
 			#OGAS removes stagnation
 			has_country_modifier = ogas
 			has_country_modifier = cybersin
-			
-			#Not planned economy
-			NOT = {
-				OR = { 
-					economic_policy = planned_economy 
-					economic_policy = planned_economy1
-				}
+		}
+
+		NOT = {
+			OR = { 
+				#Not planned economy
+				economic_policy = planned_economy 
+				economic_policy = planned_economy1
+				ruling_party_ideology = communist
+				ruling_party_ideology = socialist
```

Comment:
This section should not be separated from the OR box for ogas and cybersin

Reply:
If it is in the same section, then the game automatically removes cybersin and ogas as soon as you get the modifiers instead of just the soviet stagnation modifiers. Otherwise I'd think that the event shouldn't remove them as modifiers and just remove the soviet stagnation modifiers instead.

Reply:
I see. Yes, it would be better for the event to just remove the stagnation modifiers in that case. The purpose of this event is to remove stagnation if ogas and cybersin has been developed or the country is no longer a planned economy. Ruling party ideology should not count as a removal factor as it is only the economic policy that matters. Ogas should be removed once the economic policy is no longer planned economy because state automation is no longer needed so leaving ogas there would be an excessive benefit.

Reply:
Alright, I'll edit it accordingly then.

It might be better to make separate event for removing cybersin and ogas if the nation is not communist. Is there another event that fires that it can be tacked onto?

Reply:
**I think this event can still be used to remove the modifiers if this line is used to replace the effect for removing cybersin and ogas:**

#Remove OGAS and Cybersin modifiers if THIS is no longer a planned economy
random_owned = {
limit = { owner = { NOT = { OR = { economic_policy = planned_economy economic_policy = planned_economy1 } } } }
owner = { remove_country_modifier = ogas remove_country_modifier = cybersin }
}

**and then this line in the trigger box to replace the existing check**

OR = {
#This event fires to reduce stagnation level when OGAS or Cybersin is developed
AND = { 
OR = { has_country_modifier = ogas has_country_modifier = cybersin } 
OR = { ADD STAGNATION MODIFIERS }
}
#This event fires to remove OGAS and Cybersin modifiers if THIS no longer uses planned economy
AND = {
OR = { has_country_modifier = ogas has_country_modifier = cybersin } 
NOT = { OR = { economic_policy = planned_economy economic_policy = planned_economy1 } }
}
#This event fires to remove stagnation modifiers if THIS no longer uses planned economy
AND = {
OR = { ADD STAGNATION MODIFIERS }
NOT = { OR = { economic_policy = planned_economy economic_policy = planned_economy1 } }
}
}

**These 3 checks should cover all possible situations**

Reply:
That looks great. Added it in.